src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -1503,6 +1520,46 @@ void Replica::handle(New_principal* m)
   node->add_principal(info);
 }
 
+void Replica::handle(Network_open* m)
+{
+  std::shared_ptr<Principal> p = get_principal(m->id());
+  if (p == nullptr)
+  {
+    LOG_FAIL << "recevied network open from unknown principal, id:" << m->id()
+             << std::endl;
+  }
+
+  if (p->received_network_open_msg())
+  {
+    LOG_FAIL << "recevied network open from, id:" << m->id() << "already"
```

Comment:
do we want to continue here if we have already received a network open message from this principal?

Reply:
I guess it doesn't really make a difference..